cabal.project

```diff
@@ -1,5 +1,5 @@
 -- Bump this if you need newer packages
-index-state: 2022-02-09T00:00:00Z
+index-state: 2022-05-12T00:00:00Z
```

Comment:
bumping this index state must have caused hedgehog to be updated which leads to this error:

```
Building test suite 'plutus-ledger-api-test' for plutus-ledger-api-1.0.0.0..
[1 of 6] Compiling Spec.Builtins    ( test/Spec/Builtins.hs, dist/build/plutus-ledger-api-test/plutus-ledger-api-test-tmp/Spec/Builtins.o )
[2 of 6] Compiling Spec.CostModelParams ( test/Spec/CostModelParams.hs, dist/build/plutus-ledger-api-test/plutus-ledger-api-test-tmp/Spec/CostModelParams.o )
[3 of 6] Compiling Spec.Eval        ( test/Spec/Eval.hs, dist/build/plutus-ledger-api-test/plutus-ledger-api-test-tmp/Spec/Eval.o )
[4 of 6] Compiling Spec.Interval    ( test/Spec/Interval.hs, dist/build/plutus-ledger-api-test/plutus-ledger-api-test-tmp/Spec/Interval.o )

test/Spec/Interval.hs:93:7: error: [-Wdeprecations, -Werror=deprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
   |
93 |     , testProperty "interval intersection" intvlIntersection
   |       ^^^^^^^^^^^^

test/Spec/Interval.hs:94:7: error: [-Wdeprecations, -Werror=deprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
   |
94 |     , testProperty "interval intersection with always/never" intvlIntersectionWithAlwaysNever
   |       ^^^^^^^^^^^^

test/Spec/Interval.hs:95:7: error: [-Wdeprecations, -Werror=deprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
   |
95 |     , testProperty "interval isEmpty" intvlIsEmpty
   |       ^^^^^^^^^^^^

test/Spec/Interval.hs:96:7: error: [-Wdeprecations, -Werror=deprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
   |
96 |     , testProperty "interval overlaps" intvlOverlaps
   |       ^^^^^^^^^^^^
[5 of 6] Compiling Spec.NoThunks    ( test/Spec/NoThunks.hs, dist/build/plutus-ledger-api-test/plutus-ledger-api-test-tmp/Spec/NoThunks.o )

<no location info>: error: [-Wunused-packages, -Werror=unused-packages]
    The following packages were specified via -package or -package-id flags,
    but were not needed for compilation:
      - tasty-quickcheck-0.10.2```
      - 
Perhaps turn off -Wdeprecations in that file with an OPTIONS_GHC pragma

Reply:
Actually there are way more places that we use `testProperty`. We have to do something about it

Reply:
Looking at this commit https://github.com/qfpl/tasty-hedgehog/commit/2cba6a8c43e57dffce3f8a86eeedf9987e3e836a , it means that we have to go to *all* calls of Tasty.Hedgehog.testProperty and add a ("propertyname":: PropertyName using OverloadedStrings , like TestName) .

Reply:
I ran into this issue before when updating another dependency (this time by accident). I'm happy to fix this, since the change is largely mechanical.